engine/src/test/java/org/destinationsol/systems/DamageSystemTests/NonNegativeHealthTest.java

```diff
@@ -37,23 +39,23 @@
     public void setUp() throws Exception {
         moduleManager = new ModuleManager();
         moduleManager.init();
-        entitySystemManager = new EntitySystemManager(moduleManager.getEnvironment(), new ComponentManager());
+        entitySystemManager = new EntitySystemManager(moduleManager.getEnvironment(), new ComponentManager(), new ContextImpl());
     }
 
     @Test
     public void testDamageDoesntMakeHealthBecomeNegative() {
+        Health health;
         EntityRef entity = entitySystemManager.getEntityManager().createEntity(new Health());
-        if (entity.getComponent(Health.class).isPresent()) {
-            Health health = entity.getComponent(Health.class).get();
-            health.maxHealth = 50;
-            health.currentHealth = 50;
-            entity.setComponent(health);
-        }
+        health = entity.getComponent(Health.class).get();
+        health.maxHealth = 50;
+        health.currentHealth = 50;
+        entity.setComponent(health);
+
         DamageEvent event = new DamageEvent(60);
 
         entitySystemManager.sendEvent(event, new Health());
 
-        assertEquals(0, entity.getComponent(Health.class).get().currentHealth);
+        assertEquals(0, health.currentHealth);
```

Comment:
Isn't this effectively an unspecified behavior? Since at this point the entity doesn't exist anymore?

Reply:
That's a good point

Reply:
Might there be a situation where an entity's health could drop to zero, but not be removed? For example, if someone designs a component that lets a player live for a few seconds after their health drops to zero, so that if they have a repair kit, they can still recover.

I know that that is niche, but I think that the removal is actually separate from the health handling. If a system stops the entity from dying, then it should still have zero health.